azure-pipelines/build.yml

```diff
@@ -4,12 +4,19 @@ steps:
   inputs:
     useGlobalJson: true
 
+- task: DownloadSecureFile@1
+  displayName: 'Download Key Pair'
+  name: sqltoolssigningkey
+  inputs:
+    secureFile: sqlToolsKeyPair.snk
+    retryCount: 5
+
 - task: BatchScript@1
   displayName: 'Run script build.cmd'
   inputs:
     filename: build.cmd
-    arguments: '-target=all -mono'
-
+    arguments: '-target=all -mono -signAssembly=true -signAssemblyKeyFile="$(sqltoolssigningkey.secureFilePath)"'
```

Comment:
In general I don't think we should be signing the assemblies ourselves and instead using ESRP for that - see https://github.com/Azure/azure-functions-sql-extension/pull/301 for an example

Reply:
This is a requirement for strong name key (SNK) signing that applies to .NET Framework DLLs. 
It's not actually 'signing' the DLL but creating a unique identity. 

The DLLs are ofcourse signed with ESRP Codesigning later.

Reply:
ESRP does strong name signing. Look at the line the link brings you to.

Reply:
i think Cheena means this step is delay signing the DLL so ESRP can apply the full signing later.

Reply:
In general all SQL assemblies are signed with the same key and I don't see any reason we should change that.

Reply:
though afaik you can just add these to the props to do it
```
      <SignAssembly>true</SignAssembly>  
      <DelaySign>true</DelaySign>
<AssemblyOriginatorKeyFile>mykey.snk</AssemblyOriginatorKeyFile>
```

Reply:
I can enable delay sign and include the ESRP signing operation, that makes sense! 
I can address that with key alongside too.